app/inbound_number/rest.py

```diff
@@ -37,3 +47,40 @@ def get_available_inbound_numbers():
     inbound_numbers = [i.serialize() for i in dao_get_available_inbound_numbers()]
 
     return jsonify(data=inbound_numbers if inbound_numbers else [])
+
+
+@inbound_number_blueprint.route("/service/<uuid:service_id>", methods=["POST"])
+def add_inbound_number_to_service(service_id):
+    """
+    Route to add an inbound number to a service. If inbound_number_id is provided we
+    add that specific number, otherwise we add a random available number.
+    """
+    form = validate(request.get_json(), add_inbound_number_to_service_request)
+
+    if form.get("inbound_number_id"):
+        inbound_number_id = form["inbound_number_id"]
+    else:
+        try:
+            available_number = dao_get_available_inbound_numbers()[0]
+            inbound_number_id = available_number.id
+        except IndexError:
+            raise Exception("There are no available inbound numbers")
```

Comment:
nit: we could store the IndexError in a variable (eg `e`) and add `from e` here to chain the exceptions together.

Reply:
There's not a huge benefit, but it's probably more semantically appropriate. It says says "we're handling the IndexError and purposefully throwing this new error because of it", rather than "we've tried to handle the IndexError and something else has gone wrong".

In a printed stacktrace it changes the line `During handling of the above exception, another exception occurred:` to `The above exception was the direct cause of the following exception:`, which is a bit clearer.

Quick primer: https://stefan.sofa-rockers.org/2020/10/28/raise-from/

Note on nits I leave: I use them in the way they're mentioned [here](https://google.github.io/eng-practices/review/reviewer/standard.html#:~:text=Reviewers%20should%20always%20feel%20free,they%20could%20choose%20to%20ignore.): specifically, "Reviewers should _always_ feel free to leave comments expressing that something could be better, but if it’s not very important, prefix it with something like “Nit: “ to let the author know that it’s just a point of polish that they could choose to ignore."